docs/basics/primitives.md

```diff
@@ -5,30 +5,195 @@ to describe the set of types that are native to its platform, we leverage and ov
+
+  object uid extends Col[UUID]
+}
+```
+
+There is virtually no difference whatsoever in doing things one way or the other,
+the encodings and serializers/deserializers are handled by the same macro, which
+will do its magic invisibly in the background and generate all the Cassandra I/O
+code required to marshall those types back and forth across the wire and to/from Cassandra.
+
 ### The mechanism that generates primitives
 
 The physical implementation is based on the typeclass model, a very popular Scala concept, and phantom
-leverages implicit resolution to help Cassandra understand more of your Scala types. And there are a number of language
-features that in conjunction with appropriate Cassandra types we can leverage to produce seemingly native support
-for a wide variety of types.
+leverages implicit resolution to help Cassandra understand more Scala types and "prented" they are native.
 
-Several concepts have very direct translation into Cassandra:
+And there are a number of language features that in conjunction with appropriate Cassandra types we can leverage to produce seemingly native support for a wide variety of types.
 
+Several concepts have a very direct translation into Cassandra:
 
 #### Tuples
 
 A `scala.Tuple` can be directly mapped to a `Cassandra.Tuple`. The tuple concept excepts natively within Cassandra and
  we can therefore produce directly translatable mappings between a Scala type and a Cassandra type.
- 
- 
+
+Examples of such translations are visible below:
+
+ | phantom columns                      | Java/Scala type                             | Cassandra type    |
+ | ---------------                      |-------------------                          | ----------------- |
+ | Col[(String, String)]                | Tuple2[String, String]                      | tuple<text, text> |
+ | Col[(String, Int)]                   | Tuple2[String, Int]                         | tuple<text, int> |
+ | Col[(String, Int, List[BigDecimal])] | Tuple3[String, Int, List[BigDecimal]]       | tuple<text, int, frozen<list<decimal>>> |
+
+As seen in the third example, freezing of types should happen automatically during schema auto-generation, and
+this is valid for any kind of type that requires freezing, including:
+
+- Nested collection types.
+- Primary collection or tuple types.
+- UDT types in phantom pro.
+- Auto-derived primitives in phantom-pro.
+
+This also makes it possibly to derive arbitrary encodings for very complex types automatically.
+This feature is only available in phantom-pro using the `autotables` module.
+
+```scala
+import com.outworkers.phantom.auto._
+
+case class SubRecord(
+  id: UUID,
+  description: String,
+  tags: Set[String],
+  timestamp: DateTime
+)
+
+object SubRecord {
+  implicit val subRecordPrimitive: Primitive[SubRecord] = Primitive.auto[SubRecord].derive
+}
+```
+
+This would allow you to use `SubRecord` transparently as a natively Cassandra type
```

Comment:
Just:

This would allow you to use `SubRecord` transparently as a natively Cassandra type.
All necessary marshalling is generated at compile time - 0 runtime overhead. 

How:
Advanced macro machinery which enables arbitrary use of `SubRecord` as a valid column type.